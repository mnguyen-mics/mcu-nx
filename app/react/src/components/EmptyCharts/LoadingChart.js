import React from 'react';
import { Row, Col, Spin } from 'antd';

const LoadingChart = () => {

  return (
    <Row className="mcs-card-no-data">
      <Col span={24} className="">
        <Spin size="large" />
      </Col>
    </Row>
  );

};

export default LoadingChart;
