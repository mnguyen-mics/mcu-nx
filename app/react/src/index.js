import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import configureStore from './store';
import { Navigator } from './containers/Navigator';

const store = configureStore();

const micsProvider = (
  <Provider store={store}>
    <Navigator />
  </Provider>
);

render(
  micsProvider,
  document.getElementById('mcs-react-app') // eslint-disable-line no-undef
);
