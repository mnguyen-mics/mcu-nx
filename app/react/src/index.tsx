/* eslint-disable */
import React from 'react';
import { render } from 'react-dom';

import App from './App';

declare global {
  interface Window {
    require: any;
    mainRev: any;
  }
}

render(<App />, document.getElementById('mcs-react-app'), () => {
  window.require([window.mainRev], () => {
    //
  });
});
