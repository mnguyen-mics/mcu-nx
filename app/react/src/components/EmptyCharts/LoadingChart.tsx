import * as React from 'react';
import { Row, Col, Spin } from 'antd';

const LoadingChart: React.SFC = () => {
  return (
    <Row className="mcs-card-no-data">
      <Col span={24} className="">
        <Spin size="large" />
      </Col>
    </Row>
  );
};

export default LoadingChart;
