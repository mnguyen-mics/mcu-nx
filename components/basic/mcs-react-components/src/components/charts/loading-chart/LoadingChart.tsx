import * as React from 'react';
import { Row, Col, Spin } from 'antd';

export interface LoadingChartProps {
  className?: string;
}

const LoadingChart: React.SFC<LoadingChartProps> = ({ className }) => {
  return (
    <Row className={`mcs-loading-chart-no-data ${className ? className : ''}`}>
      <Col span={24}>
        <Spin size='large' />
      </Col>
    </Row>
  );
};

export default LoadingChart;
