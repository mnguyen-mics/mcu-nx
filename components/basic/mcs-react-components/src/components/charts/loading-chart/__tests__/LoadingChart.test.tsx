import 'jest';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import LoadingChart, { LoadingChartProps } from '../LoadingChart';

it('Should render the LoadingChart', () => {
  const props: LoadingChartProps = {
    className: 'test',
  };
  const component = TestRenderer.create(<LoadingChart {...props} />);
  const res = component.toJSON();
  expect(res).toMatchSnapshot();
});
