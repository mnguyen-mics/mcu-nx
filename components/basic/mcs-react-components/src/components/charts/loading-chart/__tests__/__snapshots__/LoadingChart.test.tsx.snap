// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the LoadingChart 1`] = `
<div
  className="ant-row mcs-loading-chart-no-data test"
  style={Object {}}
>
  <div
    className="ant-col ant-col-24"
    style={Object {}}
  >
    <div
      className="ant-spin ant-spin-lg ant-spin-spinning"
    >
      <span
        className="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          className="ant-spin-dot-item"
        />
        <i
          className="ant-spin-dot-item"
        />
        <i
          className="ant-spin-dot-item"
        />
        <i
          className="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>
</div>
`;
